test/unit/BPool/BPool_Unbind.tree

```diff
@@ -0,0 +1,23 @@
+BPool::Unbind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is not bound
+    │   └── it should revert
+    └── when token is bound
+        ├── when pool is finalized
+        │   └── it should revert
+        ├── when token is last on the tokens array
+        │   ├── it sets reentrancy lock
+        │   ├── it emits LOG_CALL event
+        │   ├── it calls _pushUnderlying
+        │   ├── it removes the token record
+        │   ├── it decreases the total weight
+        │   └── it pops from the array
+        └── when token is NOT last on the tokens array
+            ├── it removes the token record
+            ├── it removes the token from the array
+            ├── it keeps other tokens in the array
+            └── it updates records to point to the correct indices
```

Comment:
it also `decreases the total weight`, should we test that in a common context?

when is last: pops the array
when is not: swaps index for last and pops the array
when called: emits LOG, calls _pushUnderlying, decreases total weight...

Reply:
what I don't like about that is that there are not really three *branches* , we'd only be separating them for the comfort of dividing the asserts into categories. It feels wrong that one of the branches would have the exact same preconditions/setup as the other, while the tree suggests otherwise.

I think it's cleaner to 
- do all of the common asserts on the branch with the simplest setup
- in the branches with more complex setup, focus extra hard on the different behaviour

we could also just \*gasps\* _repeat_ the leafs/assertions for consistency and a little extra safety, in case some of the assertions are no longer 'common' later down the line

Reply:
> It feels wrong that one of the branches would have the exact same preconditions/setup as the other, while the tree suggests otherwise

hmmmm ok, i don't really like the repeating expectations, but they'll end up being comments to be checked (and not new test methods) so i'm ok with it